src/main/java/marquez/resources/JobRunResource.java

```diff
@@ -0,0 +1,147 @@
+    UUID jobRunGuid = UUID.randomUUID();
+
+    // TODO: Validate that the job run definition exists. Otherwise, throw a 404.
+    // This will be much easier to implement once JobRunDefinition is checked in.
+    try {
+      JobRun jobRun =
+          new JobRun(
+              jobRunGuid,
+              new Timestamp(System.currentTimeMillis()),
+              null,
+              null,
+              request.getJobRunDefinitionGuid(),
+              request.getState());
+      dao.insert(jobRun);
+
+      CreateJobRunResponse res = new CreateJobRunResponse(jobRunGuid);
+      String jsonRes = mapper.writeValueAsString(res);
+      return Response.status(Response.Status.CREATED)
+          .entity(jsonRes)
+          .type(APPLICATION_JSON)
+          .build();
+    } catch (Exception e) {
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+          .entity(Entity.json("{'error' : 'an unexpected error occurred.'}"))
+          .type(APPLICATION_JSON)
+          .build();
+    }
+  }
+
+  @PUT
+  @Consumes(APPLICATION_JSON)
+  @Timed
+  @Path("/{guid}")
+  public Response update(@PathParam("guid") final UUID guid, @Valid UpdateJobRunRequest request) {
+    try {
+      JobRun existingJobRunRow = dao.findJobRunById(guid);
+      Integer oldState = existingJobRunRow.getCurrentState();
+      Integer newState;
+      try {
+        newState = toInt(JobRunState.State.valueOf(request.getState()));
+      } catch (IllegalArgumentException iae) {
+        return Response.status(Response.Status.BAD_REQUEST)
+            .type(APPLICATION_JSON)
+            .entity(Entity.json("'{'error': 'unknown state value'}"))
+            .build();
+      }
+
+      if (!JobRun.isValidJobTransition(oldState, newState)) {
+        return Response.status(Response.Status.FORBIDDEN)
+            .type(APPLICATION_JSON)
+            .entity(Entity.json("{'error' : 'invalid transition'}"))
+            .build();
+      }
+
+      Timestamp startedAt = existingJobRunRow.getStartedAt();
+      Timestamp endedAt = existingJobRunRow.getEndedAt();
+
+      if (fromInt(oldState) == JobRunState.State.NEW
+          && fromInt(newState) == JobRunState.State.RUNNING) {
+        startedAt = new Timestamp(System.currentTimeMillis());
+      } else if (fromInt(newState).isFinished()) {
+        endedAt = new Timestamp(System.currentTimeMillis());
+      }
+
+      JobRun updatedJobRunRow =
+          new JobRun(
+              existingJobRunRow.getGuid(),
+              existingJobRunRow.getCreatedAt(),
+              startedAt,
+              endedAt,
+              existingJobRunRow.getJobRunDefinitionGuid(),
+              toInt(JobRunState.State.valueOf(request.getState())));
+
+      dao.update(updatedJobRunRow);
+
+      UpdateJobRunResponse res = new UpdateJobRunResponse(guid);
+      String jsonRes = mapper.writeValueAsString(res);
+      return Response.status(Response.Status.ACCEPTED)
+          .entity(jsonRes)
```

Comment:
TODO(code review): test out if we can get away with specifying `res` here and having it map.